### PR TITLE
Ladybird: Respect window.open() features in Qt chrome

### PR DESCRIPTION
**Pull request by ADKaster**
We can now `window.open("https://example.com", "hi", "popup=1")` and have the Qt chrome open a new window for the opened window. But, ofc, only if popups are disabled. In the future we should probably handle popups on a per-origin basis like other browsers.

### CODE REVIEW

File: Userland/Libraries/LibWeb/CMakeLists.txt [hunk 1]
```diff
@@ -429,6 +429,7 @@ set(SOURCES
     HTML/TextMetrics.cpp
     HTML/Timer.cpp
     HTML/TimeRanges.cpp
+    HTML/TokenizedFeatures.cpp
     HTML/ToggleEvent.cpp
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/CMakeLists.txt:433]: :warning: alphabetical pls

File: Userland/Libraries/LibWeb/HTML/WebViewHints.cpp [hunk 2]
```diff
@@ -6,7 +6,122 @@
...
+    auto screen_rect = page.web_exposed_screen_area();
+
+    // 5. If tokenizedFeatures["left"] exists:
+    if (auto left = tokenized_features.get("left"sv); left.has_value()) {
+        // 1. Set x to the result of invoking the rules for parsing integers on tokenizedFeatures["left"].
+        // 2. If x is an error, set x to 0.
+        x = parse_integer(*left).value_or(0);
+
+        // 3. Optionally, clamp x in a user-agent-defined manner so that the window does not move outside the Web-exposed available screen area.
+        x = min(*x, screen_rect.x());
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/HTML/WebViewHints.cpp:50]: :thinking: wonder if we should also make sure it doesn't go off the right side of the screen, i.e. limit to `screen_width - screen_x`. Similar with `y` below.

**Review by trflynn89 — COMMENTED** [review]
LGTM, just a couple comments

File: Userland/Libraries/LibWeb/HTML/WebViewHints.cpp (hunk 2 above)
Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/WebViewHints.cpp:50]: I do wonder, how much of this is going to be handled by the GUI framework on the UI process side in the end? but I can try to hit detect the sides of the box in the size cases and adjust x/y.

Comment (trflynn89) [reply Userland/Libraries/LibWeb/HTML/WebViewHints.cpp:50]: Yeah good point, I dunno - my main concern was a website throwing the window wayyy off the right of the screen where users can't get it, so I think we're good as long as we prevent that, then let the UI kits handle whatever else they want

File: Ladybird/Qt/BrowserWindow.cpp [hunk 3]
```diff
@@ -910,6 +918,31 @@ void BrowserWindow::update_displayed_zoom_level()
     m_current_tab->update_reset_zoom_button();
 }
 
+void BrowserWindow::set_window_rect(Optional<Web::DevicePixels> x, Optional<Web::DevicePixels> y, Optional<Web::DevicePixels> width, Optional<Web::DevicePixels> height)
+{
+    auto current_geometry = geometry();
+    current_geometry.moveTopLeft(mapToGlobal(current_geometry.topLeft()));
+    dbgln("current_geometry: {} {} {} {}", current_geometry.x(), current_geometry.y(), current_geometry.width(), current_geometry.height());
```
Comment (trflynn89) [inline Ladybird/Qt/BrowserWindow.cpp:925]: Looks like a whole bunch of debugging got accidentally committed here

Comment (ADKaster) [reply Ladybird/Qt/BrowserWindow.cpp]: Somehow ~~Palpatine returned~~ my changes from my macOS testing didn't get pushed up, should be all fixed now.

**Pull request merged**